pkg/utils/utils.go

```diff
@@ -288,3 +293,58 @@ func readConfigMap(path string) ([]string, error) {
 
 	return images, nil
 }
+
+func ReadCollectScanPipe(ctx context.Context) ([]eraserv1alpha1.Image, error) {
+	_, cancel := context.WithCancel(ctx)
```

Comment:
For the cancellation here we shouldn't need to create a new context.
Instead the passed in context needs to be integrated into that for loop.

Something like:

```go
timer := time.NewTimer(time.Second)
if !timer.Stop() {
    <-timer.C
}
defer timer.Stop()
for {
    f, err = os.OpenFile(...)
   // error handling
  timer.Start(time.Second)
  select {
  case <-ctx.Done():
      return ctx.Err()
  case <-timer.C:
  }
}
```

This allows the caller to cancel the context in case it's taking too long to open the file.

Reply:
timer didn't have a Start() so I used Reset()  instead